pkg/client-nfd/compat/node-validator/node-validator.go

```diff
@@ -96,91 +97,100 @@ func (nv *nodeValidator) Execute(ctx context.Context) ([]*CompatibilityStatus, e
 	return compats, nil
 }
 
-func evaluateRuleStatus(rule *nfdv1alpha1.Rule, matchStatus *nodefeaturerule.MatchStatus) ProcessedRuleStatus {
-	var matchedFeatureTerms nfdv1alpha1.FeatureMatcher
+func (nv *nodeValidator) evaluateRuleStatus(rule *nfdv1alpha1.Rule, matchStatus *nodefeaturerule.MatchStatus) ProcessedRuleStatus {
 	out := ProcessedRuleStatus{Name: rule.Name, IsMatch: matchStatus.IsMatch}
 
-	evaluateFeatureMatcher := func(featureMatcher, matchedFeatureTerms nfdv1alpha1.FeatureMatcher) []MatchedExpression {
-		out := []MatchedExpression{}
-		for _, term := range featureMatcher {
-			if term.MatchExpressions != nil {
-				for name, exp := range *term.MatchExpressions {
-					isMatch := false
-
-					// Check if the expression matches
-					for _, processedTerm := range matchedFeatureTerms {
-						if term.Feature != processedTerm.Feature || processedTerm.MatchExpressions == nil {
-							continue
-						}
-						pexp, ok := (*processedTerm.MatchExpressions)[name]
-						if isMatch = ok && exp.Op == pexp.Op && slices.Equal(exp.Value, pexp.Value); isMatch {
-							break
-						}
-					}
-
-					out = append(out, MatchedExpression{
-						Feature:     term.Feature,
-						Name:        name,
-						Expression:  exp,
-						MatcherType: MatchExpressionType,
-						IsMatch:     isMatch,
-					})
-				}
-			}
+	matchedFeatureTerms := nfdv1alpha1.FeatureMatcher{}
+	if matchStatus.MatchFeatureStatus != nil {
+		matchedFeatureTerms = matchStatus.MatchFeatureStatus.MatchedFeaturesTerms
+	}
```

Comment:
nit: just caught my eye, small readability improvement (imo)

```suggestion
	if m := matchStatus.MatchFeatureStatus; m != nil {
		matchedFeatureTerms = m.MatchedFeaturesTerms
	}
```

Reply:
Fixed.